internal/graph/track_check_resolver_test.go

```diff
@@ -0,0 +1,280 @@
+
+		tc := &TrackerCheckResolver{}
+		require.False(t, tc.validate())
+
+		tc = &TrackerCheckResolver{ctx: context.Background()}
+		require.False(t, tc.validate())
+
+		userType := trackChecker.userType("group:1#member")
+		require.Equal(t, "userset", userType)
+
+		userType = trackChecker.userType("user:ann")
+		require.Equal(t, "user", userType)
+
+		userType = trackChecker.userType("user:*")
+		require.Equal(t, "userset", userType)
+
+		r := resolutionNode{tm: time.Now().Add(-trackerInterval)}
+		require.True(t, r.expired())
+
+		r = resolutionNode{tm: time.Now().Add(trackerInterval)}
+		require.False(t, r.expired())
+	})
+
+	t.Run("tracker_prints_and_delete_path", func(t *testing.T) {
+		r := &ResolveCheckRequest{
+			StoreID:              ulid.Make().String(),
+			AuthorizationModelID: ulid.Make().String(),
+			TupleKey:             tuple.NewTupleKey("document:abc", "viewer", "user:somebody"),
+			RequestMetadata:      NewCheckRequestMetadata(20),
+		}
+		value, ok := trackChecker.loadModel(r)
+		require.NotNil(t, value)
+		require.False(t, ok)
+
+		path := "group:1#member@user"
+		sm := &sync.Map{}
+		sm.Store(
+			path,
+			&resolutionNode{
+				tm:   time.Now().Add(-trackerInterval),
+				hits: &atomic.Uint64{},
+			},
+		)
+
+		trackChecker.nodes.Store(trackerKey{store: ulid.Make().String(), model: ulid.Make().String()}, sm)
+		trackChecker.logExecutionPaths(false)
+
+		_, ok = sm.Load(path)
+		require.False(t, ok)
+	})
+
+	t.Run("tracker_success_launch_flush", func(t *testing.T) {
+		wg := sync.WaitGroup{}
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			time.Sleep(time.Duration(10) * time.Millisecond)
+		}()
+
+		path := "group:1#member@user"
+		sm := &sync.Map{}
+		sm.Store(
+			path,
+			&resolutionNode{
+				tm:   time.Now().Add(-trackerInterval),
+				hits: &atomic.Uint64{},
+			},
+		)
+
+		trackChecker.nodes.Store(trackerKey{store: ulid.Make().String(), model: ulid.Make().String()}, sm)
+
+		trackChecker.ticker.Reset(time.Duration(2) * time.Millisecond)
+		trackChecker.launchFlush()
+
+		wg.Wait()
+
+		_, ok := sm.Load(path)
+		require.False(t, ok)
+	})
+
+	t.Run("success_loadModel_and_loadPaths", func(t *testing.T) {
+		r := &ResolveCheckRequest{
+			StoreID:              ulid.Make().String(),
+			AuthorizationModelID: ulid.Make().String(),
+			TupleKey:             tuple.NewTupleKey("document:abc", "viewer", "user:somebody"),
+			RequestMetadata:      NewCheckRequestMetadata(20),
+		}
+		value, ok := trackChecker.loadModel(r)
+		require.NotNil(t, value)
+		require.False(t, ok)
+
+		path := trackChecker.getTK(tuple.NewTupleKey("folder:test", "reader", "user:somebody"))
+		trackChecker.loadPath(value, path)
+		trackChecker.addPathHits(r)
+
+		paths, ok := value.(*sync.Map)
+		require.True(t, ok)
+		_, ok = paths.Load(path)
+		require.True(t, ok)
+
+		paths, ok = value.(*sync.Map)
+		require.True(t, ok)
+		_, ok = paths.Load(path)
+		require.True(t, ok)
+	})
+
+	t.Run("success_limiter_disallow", func(t *testing.T) {
```

Comment:
I'm not sure how the code for this test achieves the goal in the name of this test, should you not assert that zero logs were emmited?